sycl/include/CL/sycl/detail/image_accessor_util.hpp

```diff
@@ -629,12 +589,13 @@ vec<ChannelType, 4> convertWriteData(vec<cl_float, 4> WriteData,
     break;
   case image_channel_type::unorm_int_101010:
     // min(convert_ushort_sat_rte(f * 1023.0f), 0x3ff)
-    Temp = multiplyEachElement(WriteData, 1023.0f);
+    Temp = WriteData * 1023.0f;
     TempInInt = Temp.convert<int, rounding_mode::rte>();
     MinVal = min_v<cl_ushort>();
     MaxVal = max_v<cl_ushort>();
     TempInIntSaturated = cl::sycl::clamp(TempInInt, MinVal, MaxVal);
-    TempInIntSaturated = cl::sycl::min(TempInIntSaturated, static_cast<int>(0x3ff));
+    TempInIntSaturated =
+        cl::sycl::min(TempInIntSaturated, static_cast<int>(0x3ff));
```

Comment:
Why you need a cast to int here?
static_assert(std::is_same<decltype(0x3ff), int>::value, "");

Reply:
you are right.  
std::cout << "datatype of 0x3FF is: " << std::is_same<decltype(0x3FF), int>::value <<std::endl;
printed a 1.
Because I had to convert the values for the other functions for template resolution, i thought i need to do this too, and didn't realise the value is an integer already.
Thanks.